etc/runonce-entries

```diff
@@ -62,6 +62,30 @@ runonce <<"EOF"
   refresh_all_pkgapp_status
 EOF
 
+# remove deprecated apps (only when running in a visible terminal)
```

Comment:
Presumably the importance of running this in a visible terminal is to ensure this runonce only runs when the user is already installing an app and a terminal is visible to display the progress of uninstalling deprecated apps?

Reply:
Run once runs as part of set-status. That happens when pi-apps opens in a background terminal, as well as during any update, install, or uninstall in a foreground terminal.

There are two reasons it needs to be only in foreground.

Background terminals cannot and should not be used since the user won't see any terminal output from any uninstalls executed from deprecated apps. 

In addition, if sudo is needed and the user does not have passwordless sudo (so any normal Linux distro that isn't piOS) the runonce would hang in the background waiting for input in a terminal that doesn't have any way for the user to enter a password.
sudo_popup isn't an option since it uses pkexec which has its own set of environment variables (clears local environment) and would break many scripts. Also every sudo prompt would ask for the password which would not be ideal.